### PR TITLE
Fix crash on no service accounts

### PR DESCRIPTION
**Pull request by donomii**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/2731

**Special notes for your reviewer**:

List service account api call crashes if no service accounts present

### CODE REVIEW

File: pkg/services/sqlstore/org_users.go [hunk 1]
```diff
@@ -117,7 +116,7 @@ func (ss *SQLStore) GetOrgUsers(ctx context.Context, query *models.GetOrgUsersQu
 	// service accounts table in the modelling
 	whereConditions = append(whereConditions, fmt.Sprintf("%s.is_service_account = %t", x.Dialect().Quote("user"), query.IsServiceAccount))
 
-	if ss.Cfg.FeatureToggles["accesscontrol"] {
+	if ss.Cfg.FeatureToggles["accesscontrol"] && query.User != nil {
```
Comment (eleijonmarck) [inline pkg/services/sqlstore/org_users.go:119]: hmm, interesting fix here. So we get a query which does not have a user I see.
But this is also related to "access control" not checking for a user right

File: build.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build ignore
```
Comment (eleijonmarck) [inline build.go:1]:
```suggestion
```

File: pkg/services/serviceaccounts/tests/common.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ func (s *ServiceAccountsStoreMock) ListTokens(ctx context.Context, orgID int64,
 	s.Calls.ListTokens = append(s.Calls.ListTokens, []interface{}{ctx, orgID, serviceAccount})
 	return nil, nil
 }
-func (s *ServiceAccountsStoreMock) ListServiceAccounts(ctx context.Context, orgID int64) ([]*models.OrgUserDTO, error) {
+func (s *ServiceAccountsStoreMock) ListServiceAccounts(ctx context.Context, orgID int64, serviceAccount int64) ([]*models.OrgUserDTO, error) {
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/tests/common.go:98]:
```suggestion
func (s *ServiceAccountsStoreMock) ListServiceAccounts(ctx context.Context, orgID int64, serviceAccountID int64) ([]*models.OrgUserDTO, error) {
```

File: pkg/services/serviceaccounts/serviceaccounts.go [hunk 4]
```diff
@@ -14,7 +14,7 @@ type Service interface {
 
 type Store interface {
 	CreateServiceAccount(ctx context.Context, saForm *CreateServiceaccountForm) (*models.User, error)
-	ListServiceAccounts(ctx context.Context, orgID int64) ([]*models.OrgUserDTO, error)
+	ListServiceAccounts(ctx context.Context, orgID, serviceAccountId int64) ([]*models.OrgUserDTO, error)
```
Comment (eleijonmarck) [inline pkg/services/serviceaccounts/serviceaccounts.go:17]:
```suggestion
	ListServiceAccounts(ctx context.Context, orgID, serviceAccountID int64) ([]*models.OrgUserDTO, error)
```

**Review by Jguer — APPROVED** [review]

**Comment by gamab** [comment]
Should have this been backported in 8.4.x? I had to backport [this](https://github.com/grafana/grafana/pull/44999) manually because of a conflict [there](https://github.com/grafana/grafana/pull/44999/files#diff-f4e455733d97354cc18cba952ee3bb379791493da2c9625d0128b887b1b0d481R121)

**Comment by eleijonmarck** [comment]
Hmm 🤔 could you nudge me tommorow?

**Pull request merged**